src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -913,12 +939,12 @@ public DuplexPipeConverter(JsonMessageFormatter jsonMessageFormatter)
             public override IDuplexPipe? ReadJson(JsonReader reader, Type objectType, IDuplexPipe? existingValue, bool hasExistingValue, JsonSerializer serializer)
             {
                 int? tokenId = JToken.Load(reader).Value<int?>();
-                return this.jsonMessageFormatter.duplexPipeTracker.GetPipe(tokenId);
+                return this.jsonMessageFormatter.duplexPipeTracker!.GetPipe(tokenId);
             }
 
             public override void WriteJson(JsonWriter writer, IDuplexPipe? value, JsonSerializer serializer)
             {
-                var token = this.jsonMessageFormatter.duplexPipeTracker.GetToken(value);
+                var token = this.jsonMessageFormatter.duplexPipeTracker!.GetToken(value);
```

Comment:
Just out of curiosity, how does the ! operator work? I couldn't find anything online about it. (Just trying to learn something here)

Reply:
I should have used my new `DuplexPipeTracker` property here. I'll fix that in a follow-up PR.

The `!` is the ["null forgiving operator"](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving).